### PR TITLE
Implement mFirstTrue evaluator on Power for Byte type

### PR DESCRIPTION
**Pull request by gita-omr**
- Returns the index of the first mask lane that is set.
- Returns vector length if none of them are set.

### CODE REVIEW

**Comment by gita-omr** [comment]
Made WIP to double check if byte reverse is needed on LE or not.

**Comment by zl-wang** [comment]
i will review when it is opened up (out of WIP).

**Comment by gita-omr** [comment]
Added description.

**Comment by gita-omr** [comment]
Improved assertions.

**Comment by gita-omr** [comment]
Removing WIP since confirmed that no reversal is needed. This is because we do use `lxvb16x` to load Byte vectors.

File: compiler/p/codegen/OMRTreeEvaluator.cpp [hunk 1]
```diff
@@ -919,7 +919,26 @@ OMR::Power::TreeEvaluator::mTrueCountEvaluator(TR::Node *node, TR::CodeGenerator
...
+
+   TR_ASSERT_FATAL_WITH_NODE(node, firstChild->getDataType().getVectorLength() == TR::VectorLength128,
+                             "Only 128-bit vectors are supported %s", node->getDataType().toString());
+
+   TR_ASSERT_FATAL_WITH_NODE(node, firstChild->getDataType().getVectorNumLanes() == 16,
+                             "Unsupported vector type %s for mToLongBits\n", firstChild->getDataType().toString());
+
+
+   TR::Register *srcReg = cg->evaluate(firstChild);
+   TR::Register *tmpReg = cg->allocateRegister(TR_VRF);
```
Comment (zl-wang) [inline compiler/p/codegen/OMRTreeEvaluator.cpp:932]: two lines re tmpReg can be eliminated.

File: compiler/p/codegen/OMRTreeEvaluator.cpp [hunk 2]
```diff
@@ -919,7 +919,26 @@ OMR::Power::TreeEvaluator::mTrueCountEvaluator(TR::Node *node, TR::CodeGenerator
...
+   TR::Register *resReg = cg->allocateRegister();
```
Comment (zl-wang) [inline compiler/p/codegen/OMRTreeEvaluator.cpp:933]: recommend making resReg GPR explicitly

File: compiler/p/codegen/OMRTreeEvaluator.cpp [hunk 3]
```diff
@@ -919,7 +919,26 @@ OMR::Power::TreeEvaluator::mTrueCountEvaluator(TR::Node *node, TR::CodeGenerator
...
+
+   generateTrg1Src1Instruction(cg, OMR::InstOpCode::vclzlsbb, node, resReg, srcReg);
```
Comment (zl-wang) [inline compiler/p/codegen/OMRTreeEvaluator.cpp:935]: have a general concern maybe unrelated to this in general:  `vclzlsbb` is checking `bit[7]`, while `vextract*m` are checking `bit[0]`.  they appear to be incompatible ... might be they have different types of input (lane-mask vs.  true-false array???)

**Review by zl-wang — CHANGES_REQUESTED** [review]

File: compiler/p/codegen/OMRTreeEvaluator.cpp (hunk 1 above)
Comment (gita-omr) [reply compiler/p/codegen/OMRTreeEvaluator.cpp]: Done

File: compiler/p/codegen/OMRTreeEvaluator.cpp (hunk 2 above)
Comment (gita-omr) [reply compiler/p/codegen/OMRTreeEvaluator.cpp]: Done

File: compiler/p/codegen/OMRTreeEvaluator.cpp (hunk 3 above)
Comment (gita-omr) [reply compiler/p/codegen/OMRTreeEvaluator.cpp:934]: Yes, that's true. But both fit our purposes for `mFirstTrue `and` mLongToBits` correspondingly.

Comment (gita-omr) [reply compiler/p/codegen/OMRTreeEvaluator.cpp:934]: Actually, lane mask in Vector API is a true-false array essentially.

**Comment by gita-omr** [comment]
Rebased.

**Review by zl-wang — APPROVED** [review]
looks good now.

**Comment by 0xdaryl** [comment]
Jenkins build plinux,aix

**Pull request merged**